pkg/storaged/anaconda.jsx

```diff
@@ -19,10 +19,85 @@
 
 import client from "./client.js";
 
+import { decode_filename } from "./utils.js";
+import { parse_subvol_from_options } from "./btrfs/utils.jsx";
+
 export const AnacondaAdvice = () => {
     if (!client.in_anaconda_mode())
         return null;
 
     // Nothing yet.
     return null;
 };
+
+export function export_mount_point_mapping() {
+    if (!client.in_anaconda_mode())
+        return;
+
+    function fstab_info(config) {
+        let dir;
+        let subvols;
+
+        for (const c of config) {
+            if (c[0] == "fstab") {
+                const d = client.strip_mount_point_prefix(decode_filename(c[1].dir.v));
+                if (d) {
+                    const sv = parse_subvol_from_options(decode_filename(c[1].opts.v));
+                    if (sv) {
+                        if (sv.pathname) {
+                            if (!subvols)
+                                subvols = { };
+                            subvols[sv.pathname] = { dir: d };
+                        }
+                    } else if (!dir) {
+                        dir = d;
+                    }
+                }
+            }
+        }
+
+        if (dir || subvols)
+            return {
+                type: "filesystem",
+                dir,
+                subvolumes: subvols
+            };
+    }
+
+    function block_info(block) {
+        if (block.IdUsage == "filesystem") {
+            return fstab_info(block.Configuration);
+        } else if (block.IdUsage == "swap") {
+            return {
+                type: "swap",
+            };
+        } else if (block.IdUsage == "crypto") {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19923-20240202-122001-15d323ee-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)